api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -734,3 +737,13 @@ message CustomInlineHeader {
   // The type of the header that is expected to be set as the inline header.
   InlineHeaderType inline_header_type = 2 [(validate.rules).enum = {defined_only: true}];
 }
+
+message MemoryAllocatorManager {
+  // Configures tcmalloc to perform background release of free memory in amount of bytes per second;
+  uint64 bytes_to_release = 1 [(validate.rules).uint64 = {gte: 1}];
+
+  // Interval in milliseconds for memory releasing. If specified, during every
+  // interval Envoy will try to ``release bytes_to_release`` of free memory back to operating system for reuse.
+  // Defaults to 1000 milliseconds.
+  google.protobuf.Duration memory_release_interval = 2;
```

Comment:
Another idea is maybe we can add admin endpoint for release the memory. Then the control-plane can collect the memory usage and pick a time to free the memory when the workload is low. But this is separated PR or issue to discuss.